File: src/main/kotlin/tsp/graphic/LineComponent.java
package tsp.graphic;

import tsp.Gene;

import javax.swing.*;
import java.awt.*;
import java.awt.geom.Line2D;
import java.util.ArrayList;

class LineComponent extends JComponent {
    private ArrayList<Line2D.Double> lines;

    LineComponent(int width, int height) {
        setPreferredSize(new Dimension(width, height));
        lines = new ArrayList<>();
    }

    void addLine(Gene c1, Gene c2) {
        Line2D.Double line = new Line2D.Double(c1.getX(), c1.getY(), c2.getX(), c2.getY());
        lines.add(line);
    }

    public void paintComponent(Graphics g) {
        g.setColor(Color.white);
        g.fillRect(0, 0, getWidth(), getHeight());
        g.setColor(Color.black);
        for (Line2D.Double line : lines) {
            g.drawLine(
                    (int) line.getX1(),
                    (int) line.getY1(),
                    (int) line.getX2(),
                    (int) line.getY2()
            );
        }
    }
}

File: src/main/kotlin/graphic/LineComponent.kt
package graphic

import Gene

import javax.swing.*
import java.awt.*
import java.awt.geom.Line2D
import java.util.ArrayList

internal class LineComponent(width: Int, height: Int) : JComponent() {
    private val lines: MutableList<Line2D.Double>

    init {
        preferredSize = Dimension(width, height)
        lines = mutableListOf()
    }

    fun addLine(c1: Gene, c2: Gene) {
        val line = Line2D.Double(c1.x.toDouble(), c1.y.toDouble(), c2.x.toDouble(), c2.y.toDouble())
        lines.add(line)
    }

    public override fun paintComponent(g: Graphics) {
        g.color = Color.white
        g.fillRect(0, 0, width, height)
        g.color = Color.black
        for (line in lines) {
            g.drawLine(
                line.getX1().toInt(),
                line.getY1().toInt(),
                line.getX2().toInt(),
                line.getY2().toInt()
            )
        }
    }
}